drivers/peci/peci_handlers.c

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/peci.h>
+#include <syscall_handler.h>
+
+Z_SYSCALL_HANDLER(peci_config, dev, bitrate)
+{
+	return z_impl_peci_config((struct device *)dev, (u16_t)bitrate);
+}
+
+Z_SYSCALL_HANDLER(peci_enable, dev)
+{
+	return z_impl_peci_enable((struct device *)dev);
+}
+
+Z_SYSCALL_HANDLER(peci_disable, dev)
+{
+	return z_impl_peci_disable((struct device *)dev);
+}
+
+Z_SYSCALL_HANDLER(peci_transfer, dev, msg)
```

Comment:
This isn't going to work as written.

The API is passing the struct peci_msg *by value* which is not going to work across a system call, all system call args have to be at most register width. You have to pass it by reference.

In this function, you will then need to validate the peci_msg, including its sub-members, in particular the read/write buffer members.